src/python/PSetTweaks/WMTweak.py

```diff
@@ -518,3 +520,91 @@ def makeOutputTweak(outMod, job):
     #      output module based on the settings in the section
 
     return result
+
+
+def readAdValues(attrs, adname, castInt=False):
+    """
+    A very simple parser for the ads available at runtime.  Returns
+    a dictionary containing
+    - attrs: A list of string keys to look for.
+    - adname: Which ad to parse; "job" for the $_CONDOR_JOB_AD or
+      "machine" for $_CONDOR_MACHINE_AD
+    - castInt: Set to True to force the values to be integer literals.
+      Otherwise, this will return the values as a string representation
+      of the ClassAd expression.
+
+    Note this is not a ClassAd parser - will not handle new-style ads
+    or any expressions.
+
+    Will return a dictionary containing the key/value pairs that were
+    present in the ad and parseable.
+
+    On error, returns an empty dictionary.
+    """
+    adfile = None
+    if adname == 'job':
+        adfile = os.environ.get("_CONDOR_JOB_AD")
+    elif adname == 'machine':
+        adfile = os.environ.get("_CONDOR_MACHINE_AD")
+    else:
+        print("Invalid ad name requested for parsing: %s" % adname)
+        return {}
```

Comment:
Actually - I think I want both, since the two error cases are somewhat distinct.